openmls/src/group/core_group/new_from_welcome.rs

```diff
@@ -136,21 +122,56 @@ impl CoreGroup {
                 },
             };
 
-        // Commit secret is ignored when joining a group, since we already have
-        // the joiner_secret.
-        let (tree, _commit_secret_option) = TreeSync::from_nodes_with_secrets(
+        let tree = TreeSync::from_nodes(
             backend,
             ciphersuite,
             &nodes,
-            verifiable_group_info.signer(),
-            path_secret_option,
-            encryption_key_pair,
+            key_package_bundle
+                .key_package()
+                .leaf_node()
+                .encryption_key(),
         )
         .map_err(|e| match e {
             TreeSyncFromNodesError::LibraryError(e) => e.into(),
             TreeSyncFromNodesError::PublicTreeError(e) => WelcomeError::PublicTreeError(e),
         })?;
 
+        let diff = tree.empty_diff();
+
+        // Read the encryption key pair from the key store and delete it there.
+        let leaf_keypair = EncryptionKeyPair::read_from_key_store(
```

Comment:
Let's do keystore operation at the beginning of the function if possible. This will make it easier splitting them out later.